.github/workflows/golang-build-test.yml

```diff
@@ -0,0 +1,88 @@
+---
+name: Golang Build & Test
+on:  # yamllint disable-line rule:truthy
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+
+jobs:
+  pre_job_src_go_determinator:
+    runs-on: ubuntu-latest
+    # Map a step output to a job output
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@master
+        with:
+          # Only run the main job for changes including the following paths
+          paths: '[".github/workflows/golang-build-test.yml", "src/go/**"]'
+
+  build_src_go:
+    if: ${{ needs.pre_job_src_go_determinator.outputs.should_skip != 'true' }}
+    strategy:
+      matrix:
+        go-version: [1.17.x]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Test
+        run: |
+          cd src/go/
+          go build ./...
+
+  test_src_go:
+    if: ${{ needs.pre_job_src_go_determinator.outputs.should_skip != 'true' }}
+    strategy:
+      matrix:
+        go-version: [1.17.x]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Setup gotestsum
+        uses: autero1/action-gotestsum@v1.0.0
+        with:
+          gotestsum_version: 1.7.0
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Test
+        run: |
+          cd src/go/
+          gotestsum --format=testname -- -race ./...
```

Comment:
You can add `--junitfile test_src_go.xml` and then upload that result like this:
https://github.com/magma/magma/blob/master/.github/workflows/cloud-workflow.yml#L151

Reply:
<img width="994" alt="Screen Shot 2021-09-10 at 6 00 34 AM" src="https://user-images.githubusercontent.com/37634144/132843723-275f5b12-b1eb-449a-95cf-428f63dd91ee.png">

That will trigger these auto-updating comments

Reply:
Added upload, don't yet see the auto-updated comments though? Upload log shows success.

Reply:
Hmm that's weird..... @quentinDERORY might know more on how this actually works :D 

On another note, how does this work the test being run multiple times in different platforms? I wonder if there's a collision of test results if the .xml files are all named the same?

Reply:
I wondered if the multiple runs would cause a problem, yes. I had thought it might be smart enough but wasn't sure.

Reply:
You need to add the name of your workflow there: https://github.com/magma/magma/blob/1861aeef6c348f845fb76bc3a62d6835d3757946/.github/workflows/unit-test-workflow.yml#L6
The upload artifact only upload github action artifacts, this second job does the actual comment on the PR.
This has to do again with github token permission on pull request on forked PR.

Reply:
ah there you go! it is the pull_request_target magic again! :D